flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java

```diff
@@ -248,6 +253,45 @@ public void testNoRestart() throws Exception {
 		assertEquals(JobStatus.FAILED, eg.getState());
 	}
 
+	/**
+	 * Tests that the execution of the restart logic of the failover strategy is dependent on the restart strategy
+	 * calling {@link RestartCallback#triggerFullRecovery()}.
+	 */
+	@Test
+	public void testFailoverExecutionDependentOnRestartStrategyRecoveryTrigger() throws Exception {
+		final JobGraph jobGraph = createBatchJobGraph();
+		final CompletableFuture<Void> restartCallFuture = new CompletableFuture<>();
+		final RestartStrategy restartStrategy = new RestartStrategy() {
```

Comment:
Is it not possible to use `TestRestartStrategy manuallyTriggered()`? It has the method `getNumberOfQueuedActions()`.